src/main/java/de/muenchen/isi/domain/service/AbfrageService.java

```diff
@@ -123,19 +132,61 @@ public InfrastrukturabfrageModel patchAbfrageAngelegt(final Infrastrukturabfrage
             abfrage.getAbfrage().getDokumente(),
             originalAbfrageDb.getAbfrage().getDokumente()
         );
-        abfrage.getAbfrage().setStatusAbfrage(originalAbfrageDb.getAbfrage().getStatusAbfrage());
+        final var abfrageToSave = this.abfrageDomainMapper.request2Model(abfrage, originalAbfrageDb);
+        return this.saveInfrastrukturabfrage(abfrageToSave);
+    }
+
+    /**
+     * Diese Methode setzt eine {@link AbfragevarianteModel} auf Relevant. Diese muss sich im Status {@link StatusAbfrage#IN_BEARBEITUNG_SACHBEARBEITUNG} befinden.
+     *
+     * @param abfrageId         die Abfrage zum updaten
+     * @param abfragevarianteId die Abfragevariante welche man Relevant setzten möchte
+     * @return das geupdatete {@link InfrastrukturabfrageModel}
+     * @throws EntityNotFoundException           falls die Abfrage oder Abfragevariante nicht gefunden wurde
+     * @throws UniqueViolationException          falls es schon eine Abfragevariante Relevant ist
+     * @throws OptimisticLockingException        falls in der Anwendung bereits eine neuere Version der Entität gespeichert ist
+     * @throws AbfrageStatusNotAllowedException  fall die Abfrage den falschen Status hat
+     * @throws FileHandlingFailedException
+     * @throws FileHandlingWithS3FailedException
+     * @throws BauvorhabenNotReferencedException falls die Abfrage keinem Bauvorhaben dazugehört
+     */
+    public InfrastrukturabfrageModel setAbfragevarianteRelevant(final UUID abfrageId, final UUID abfragevarianteId)
+        throws EntityNotFoundException, UniqueViolationException, OptimisticLockingException, AbfrageStatusNotAllowedException, FileHandlingFailedException, FileHandlingWithS3FailedException, BauvorhabenNotReferencedException {
+        final var abfrage = this.getInfrastrukturabfrageById(abfrageId);
+        this.throwAbfrageStatusNotAllowedExceptionWhenStatusAbfrageIsInvalid(
+                abfrage.getAbfrage(),
+                StatusAbfrage.IN_BEARBEITUNG_SACHBEARBEITUNG
+            );
+        dokumentService.deleteDokumenteFromOriginalDokumentenListWhichAreMissingInParameterAdaptedDokumentenListe(
+            abfrage.getAbfrage().getDokumente(),
+            abfrage.getAbfrage().getDokumente()
+        );
+
+        final var abfragevariante = abfrage
+            .getAbfragevarianten()
+            .stream()
+            .filter(abfragevarianteModel -> abfragevarianteModel.getId().equals(abfragevarianteId))
+            .findFirst()
+            .orElseThrow(() -> {
+                final var message = "Abfragevariante wurde nicht gefunden";
+                log.error(message);
+                return new EntityNotFoundException(message);
+            });
+        abfragevariante.setRelevant(!abfragevariante.isRelevant());
+        this.checkForUniqueRelevantAbfragevariante(abfrage);
+        abfrage.getAbfrage().setStatusAbfrage(abfrage.getAbfrage().getStatusAbfrage());
```

Comment:
Ist das setzen des Status hier nötigt?

Der Status der in der `abfrage` vorhanden ist wird wieder in der selben `abfrage` gesetzt.